### PR TITLE
Slhmy/judge result

### PR DESCRIPTION
**Pull request by slhmy**
We should discuss whether the design is acceptable.

### CODE REVIEW

**Review by Zztrans — COMMENTED** [review]
In current judger, if a previous testdata fail, it will still run next test.

So the struct is fine, the len can be abtain by []TestPoints.

The judger may be expected to not run next data if fail previously?

Maybe a total testdata len is needed?

**Comment by slhmy** [comment]
> In current judger, if a previous testdata fail, it will still run next test.
> 
> So the struct is fine, the len can be abtain by []TestPoints.
> 
> The judger may be expected to not run next data if fail previously?
> 
> Maybe a total testdata len is needed?

Fair enough, but maybe we should use map for TestPoints? Because idx of each testcase is the filename.

File: src/service/model/judge.go [hunk 1]
```diff
@@ -4,6 +4,41 @@ import (
...
+)
+
+type JudgeResult struct {
+	MainVerdict    JudgeVerdict         `json:"verdict"`        // A merge of all TestPoints' verdict, according to the pirority
+	Detail         string               `json:"detail"`         // A brief description of the result
+	TestPointCount uint64               `json:"testPointCount"` // Won't be stored in database
+	TestPointMap   map[string]TestPoint `json:"testPoints"`     // Won't be stored in database
+	TestPointsJson string               `json:"-"`              // Used to store TestPoints in database
+	AverageTimeMs  uint64               `json:"averageTimeMs"`  // Won't be stored in database
+}
```
Comment (KanadeSiina) [inline src/service/model/judge.go:28]: Memory cost should also be considered, while MaxTime & MaxMemory are more usable than average ones.

**Review by KanadeSiina — CHANGES_REQUESTED** [review]

File: src/service/model/judge.go (hunk 1 above)
Comment (slhmy) [reply src/service/model/judge.go:28]: I‘m not sure if we have an accurate memory monitor in judger now.

**Review by KanadeSiina — APPROVED** [review]
It is enough and we can see how the judger works with memory later.

**Pull request merged**